utils/warnings.ml

```diff
@@ -514,15 +513,35 @@ let letter_alert tokens =
         else
           Format.fprintf ppf "%a%d..%d" print_modifier m a b
     | Letter(l,Some m) -> Format.fprintf ppf "%a%c" print_modifier m l
-    | Letter(l,None) -> print_char ppf l
+    | Letter(l,None) -> print_warning_char ppf l
   in
-  match List.find_map deprecated tokens with
-  | None -> None
-  | Some example ->
+  let consecutive_letters =
+    (* we are tracking consecutive unsigned letter in warning strings:
+       for instance in '-w "not-principa"'. *)
+    let commit_chunk l = function
+      | [] | [ _ ] -> l
+      | _ :: _ :: _ as chunk -> List.rev chunk :: l
+    in
+    let group_consecutive_letters (l,current) = function
+    | Letter (x, None) -> (l, x::current)
+    | _ -> (commit_chunk l current, [])
+    in
+    let l, on_going =
+      List.fold_left group_consecutive_letters ([],[]) tokens
+    in
+    commit_chunk l on_going
+  in
+  match consecutive_letters with
+  | [] | [] :: _ -> None
+  | example :: _  ->
```

Comment:
I'm not fond of the `[] :: _` case:

1. We statically know it cannot happens because the lists are non-empty (in fact (length >= 2)), so this could instead be `assert (examples <> [])` below.
2. It looks like we don't even need this insertion because the code would not blow up horribly if the list was empty.

So I would be in favor of dropping that subcase completely.

Reply:
You are right, this was a remnant of the previous iteration on the warning message.